src/test/java/TestNettyTransport.java

```diff
@@ -0,0 +1,109 @@
+import java.net.UnknownHostException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opensearch.common.component.Lifecycle;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportSettings;
+
+import transportservice.RunPlugin;
+import transportservice.netty4.Netty4Transport;
+
+public class TestNettyTransport extends OpenSearchTestCase {
+
+    private RunPlugin runPlugin;
+    private ThreadPool threadPool;
+
+    @BeforeEach
+    public void setUp() throws UnknownHostException {
+        this.runPlugin = new RunPlugin();
+        this.threadPool = new TestThreadPool("test");
+    }
+
+    // test Netty can bind to multiple ports, default and additional client
+    @Test
+    public void testNettyCanBindToMultiplePorts() throws Exception {
+
+        Settings nettySettings = Settings.builder()
+            .put("node.name", "netty_test")
+            .put(TransportSettings.BIND_HOST.getKey(), "127.0.0.1")
+            .put("transport.profiles.default.port", 0)
+            .put("transport.profiles.client1.port", 0)
+            .build();
+
+        try (Netty4Transport transport = startNettyTransport(runPlugin.getNetty(nettySettings, threadPool))) {
```

Comment:
Store the transport object returned from `getNetty ` separately and the pass it to `startNettyTransport ` as an argument. Instead of invoking a method inside a method.